internal/utils/nat.go

```diff
@@ -0,0 +1,168 @@
+package utils
+
+import (
+	"fmt"
+	"github.com/docker/machine/libmachine/log"
+	"github.com/ionos-cloud/docker-machine-driver/internal/pointer"
+	"github.com/ionos-cloud/docker-machine-driver/pkg/sdk_utils"
+	sdkgo "github.com/ionos-cloud/sdk-go/v6"
+	"golang.org/x/exp/maps"
+	"strconv"
+	"time"
+)
+
+func (c *Client) GetNats(datacenterId string) (*sdkgo.NatGateways, error) {
+	nats, _, err := c.NATGatewaysApi.DatacentersNatgatewaysGet(c.ctx, datacenterId).Depth(1).Execute()
+	if err != nil {
+		return nil, sdk_utils.ShortenOpenApiErr(err)
+	}
+	return &nats, nil
+}
+
+func (c *Client) GetNat(datacenterId, natId string) (*sdkgo.NatGateway, error) {
+	nat, _, err := c.NATGatewaysApi.DatacentersNatgatewaysFindByNatGatewayId(c.ctx, datacenterId, natId).Execute()
```

Comment:
you can add here apiResponse.LogInfo()  if you want some additional logging with the status and response time.

Reply:
the thing is, the `log` used by docker machine is a different package to the `log` used by that function. (and using -D flag which is usually what is used to debug `rancher-machine` or `docker-machine` CLIs obviously won't run our separate loggers)

perhaps, in the future, we could create a `CustomLogInfo` in our sdks, which can be passed an output func, such as this custom docker log func? Then, the func `LogInfo` can still stay, but implicitly call `CustomLogInfo` with the logger we normally use

Reply:
good point